pkg/reconciler/pipelinerun/resources/validate_params.go

```diff
@@ -110,3 +113,42 @@ func ValidateParameterTypesInMatrix(state PipelineRunState) error {
 	}
 	return nil
 }
+
+// ValidateParamArrayIndex validates if the param reference to an array param is out of bound.
+// error is returned when the array indexing reference is out of bound of the array param
+// e.g. if a param reference of $(params.array-param[2]) and the array param is of length 2.
+func ValidateParamArrayIndex(ctx context.Context, ps *v1beta1.PipelineSpec, pipelineRunParams v1beta1.Params) error {
```

Comment:
I remember moving to api pkg as a member function is a suggestion from @pritidesai and thus introduced in https://github.com/tektoncd/pipeline/pull/6180. 
I wonder if @pritidesai has any comments here? 

I think if this ValidateParamArrayIndex function is not called in validation webhook, maybe we should avoid putting it in the `_validate.go` file? So I prefer this PR's change.

Reply:
I agree; I don't think this makes sense as a member function for pipelines since it really only makes sense in the context of a pipelinerun.